src/ModelContextProtocol/Protocol/Transport/SseResponseStreamTransport.cs

```diff
@@ -39,23 +49,40 @@ void WriteJsonRpcMessageToBuffer(SseItem<IJsonRpcMessage?> item, IBufferWriter<b
         return _sseWriteTask = SseFormatter.WriteAsync(sseItems, sseResponseStream, WriteJsonRpcMessageToBuffer, cancellationToken);
     }
 
+    /// <inheritdoc/>
     public ChannelReader<IJsonRpcMessage> MessageReader => _incomingChannel.Reader;
 
+    /// <inheritdoc/>
     public ValueTask DisposeAsync()
     {
         _incomingChannel.Writer.TryComplete();
         _outgoingSseChannel.Writer.TryComplete();
         return new ValueTask(_sseWriteTask ?? Task.CompletedTask);
     }
 
-    public Task SendMessageAsync(IJsonRpcMessage message, CancellationToken cancellationToken = default) =>
-        _outgoingSseChannel.Writer.WriteAsync(new SseItem<IJsonRpcMessage?>(message), cancellationToken).AsTask();
+    /// <inheritdoc/>
+    public Task SendMessageAsync(IJsonRpcMessage message, CancellationToken cancellationToken = default)
+    {
+        if (_sseWriteTask is null)
+        {
+            throw new InvalidOperationException($"Transport is not connected. Make sure to call {nameof(RunAsync)} first.");
```

Comment:
May not matter, but this is going to be propagated out synchronously rather than as part of the returned Task.